src/ring_theory/polynomial/cyclotomic.lean

```diff
@@ -486,7 +491,7 @@ begin
       cyclotomic'_eq_X_pow_sub_one_div hpos hz, finset.prod_congr (refl k.proper_divisors) h]
 end
 
-/-- Any `n`-th primitive root of unity is a root of `cyclotomic n ℤ`.-/
+/-- Any `n`-th primitive root of unity is a root of `cyclotomic n K`.-/
 lemma is_root_cyclotomic {n : ℕ} {K : Type*} [field K] (hpos : 0 < n) {μ : K}
   (h : is_primitive_root μ n) : is_root (cyclotomic n K) μ :=
```

Comment:
I'm confused. Why does this need `K` to be a field? Shouldn't this hold over all rings, or at least all domains?

Reply:
this was just a lemma that already existed that had the wrong docstring - I'll try generalize this. but yes, my result at least gives us that it holds over domains

Reply:
okay, so I had a little dig and this basically comes down to the fact that we only have `p` splits ↔ `p = Π i : roots, (X - i)` for fields, which seems to be generalizable (the relevant results are in `field_division` and `splitting_field` so probably were just made for fields as that's what was important at the time) but not without a refactor. I think weakening to integral domain, charp may require `↑n ≠ 0` as in my new theorem (or something more fiddly) ]because of the values of `ɸₙ(1)`.

Reply:
Ok, maybe that's better left for a future PR

Reply:
cc @riccardobrasca